### DISCUSSION TITLE
PUID and PGID environment variables not working

### DISCUSSION THREAD

**Discussion by jamestford** [zwave-js/zwave-js-ui#2106] [General]
I am currently unable to set PUID and PGID environment variables to run zwavejs2mqtt as a non-root user. No matter how I set things up when running the container will create files as root and modify files as root. Any thoughts?

**Comment by robertsLando** [comment]
Are you speaking about docker? AFAIK you cannot run it without root privileges as it needs to access tty

**Reply by jamestford** [reply]
Yes, this should be easily overcome by adding the user to the group which has access to devices, for example in ubuntu it is the dial out group. As an example see how the zigbee2mqtt group has handled this here: https://www.zigbee2mqtt.io/guide/installation/02_docker.html#creating-the-initial-configuration

**Reply by robertsLando** [reply]
https://github.com/zwave-js/zwavejs2mqtt/pull/1014

**Comment by kpine** [comment]
A feature request has already been opened for this functionality, https://github.com/zwave-js/zwavejs2mqtt/issues/783. There was an attempt to implement that feature, but the PR was closed.

You might want to read through the conversations in that issue, and especially that closed PR. From my reading, it's not that simple, and there are concerns about breaking existing installations.
